packages/ui/src/page/dashboard/Dashboard.jsx

```diff
@@ -58,9 +65,40 @@ function Dashboard() {
     }
   }, [userDataResponse]);
 
+  useEffect(() => {
+    if (errorMsg) {
+      toast.error(errorMsg);
+    }
+  }, [errorMsg, toast]);
+
   const handleDeleteClick = (index) => {
     setSelectedKey(apiKeys[index]);
-    setShowDeleteModal(true);
+    setConfirmKeyName("");
+    setShowModal(true);
+  };
+
+  const handleKeyNameChange = (e) => {
+    setConfirmKeyName(e.target.value);
+  };
+
+  const handleDeleteKey = async () => {
```

Comment:
This function implies it handles action on pressing delete key on keyboard.
```suggestion
  const handleDelete = async () => {
```